### ISSUE TITLE
Resume with previous learning rate

### ISSUE THREAD

**Issue by cneyang** [microsoft/Semi-supervised-learning#50]
Currently, the resume option only loads model parameters.
In order to resume from the exact point where the training left off, saving and loading optimizer parameter is required.

**Comment by cneyang** [comment]
I see saving optimizer parameters but when I resume, the learning rate starts from the beginning.

**Comment by cneyang** [comment]
This happened because the `num_train_iter` argument was increased but started from the same checkpoint.

**Comment by Hhhhhhao** [comment]
Model parameters, optimizer parameters, scheduler parameters are all loaded during resume. 
But if one of the above parameters doesn't match, load_model function would not load the failed one and stop loading more. 
Check if you have any error reported or "fail to resume" printed out in the log. 
Besides, if you change the num_train_iter, you cannot expect the model starts from the same point because scheduler changes.
